### PR TITLE
fix(core): single-file bundling breaks due to left over temp dir

### PR DESCRIPTION
**Pull request by hoegertn**
This change fixes a bad behavior of the asset bundling if we use the SINGLE_FILE asset type with the OUTPUT hash type.
Because only the created file is moved and the temporary bundle dir is left over, subsequent bundling runs fail and create empty asset files.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by hoegertn** [comment]
Exemption Request as this does not change anything in CloudFormation. It is a change to the files that are created/removed in the assembly folder

File: packages/aws-cdk-lib/core/lib/asset-staging.ts [hunk 1]
```diff
@@ -343,11 +343,16 @@ export class AssetStaging extends Construct {
     this.stageAsset(bundledAsset.path, stagedPath, 'move');
 
     // If bundling produced a single archive file we "touch" this file in the bundling
-    // directory after it has been moved to the staging directory. This way if bundling
+    // directory after it has been moved to the staging directory if the hash is known before bundling. This way if bundling
     // is skipped because the bundling directory already exists we can still determine
     // the correct packaging type.
+    // If the hash is calculated after bundling we remove the temp dir now
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/core/lib/asset-staging.ts:349]:
```suggestion
    // If the hash is calculated after bundling we remove the temp dir now.
```

File: packages/aws-cdk-lib/core/lib/asset-staging.ts [hunk 2]
```diff
@@ -343,11 +343,16 @@ export class AssetStaging extends Construct {
     this.stageAsset(bundledAsset.path, stagedPath, 'move');
 
     // If bundling produced a single archive file we "touch" this file in the bundling
-    // directory after it has been moved to the staging directory. This way if bundling
+    // directory after it has been moved to the staging directory if the hash is known before bundling. This way if bundling
     // is skipped because the bundling directory already exists we can still determine
     // the correct packaging type.
+    // If the hash is calculated after bundling we remove the temp dir now.
```
Comment (paulhcsun) [inline packages/aws-cdk-lib/core/lib/asset-staging.ts:349]:
```suggestion
    // If the hash is calculated after bundling we remove the temporary directory now.
```

**Review by paulhcsun — APPROVED** [review]
Thanks for adding this fix @hoegertn!

**Pull request merged**